lib/suo/client/base.rb

```diff
@@ -81,7 +83,12 @@ def unlock(token)
           acquisition_lock = remove_lock(cleared_locks, token)
 
           break unless acquisition_lock
-          break if set(serialize_locks(cleared_locks), cas)
+
+          if @options[:lock_release_removes_key] && cleared_locks.empty?
```

Comment:
this seems a bit overcomplicated. i feel like we should have a new set flag (`expire`) and it should expire in some timeout to avoid thrashing. so this would look like e.g.

`set(serialize_locks(cleared_locks), cas, expire: cleared_locks.empty?)`

Reply:
Good idea, your approach is much more elegant.